pylife/materiallaws/rambgood.py

```diff
@@ -79,13 +107,90 @@ def plastic_strain(self, stress):
         -------
         strain : array-like float
             The resulting plastic strain
+        '''
+        absstress, signstress = self._get_abs_sign(stress)
+        return signstress * np.power(absstress/self._K, 1./self._n)
 
-        Raises
-        ------
-        ValueError if stress is negative
+    def _get_abs_sign(self, x):
+        '''Calculate the absolute value and the sign for a given input
+        
+        Parameters
+        ----------
+        x : array-like float
+            The input
+
+        Returns
+        -------
+        abs_x : array-like float
+            The resulting absolute value
+        sign_x : array-like float
+            The resulting sign of the input
         '''
-        self._fail_if_negative(stress)
-        return np.power(stress/self._K, 1./self._n)
+        abs_x = np.fabs(x)
+        sign_x = np.sign(x)
+        return abs_x, sign_x
+
+    def stress(self, strain, elasticstrain=False, rtol=1e-5, tol=1e-6):
```

Comment:
I wanted to give the user the functionality to calculate the stress if the elastic strain is given. In a fictitious use case I think 
```python
stress = rambgood.stress(elaststrain, elasticstrain=True)
```
is nicer than
```
stress = rambgood.E * elaststrain
```

I did not intend to introduce the parameter to calculate `stress0`.

Reply:
Then I'd rather introduce `.elastic_stress()`.  That would be consistent with `.strain()` / `.elastic_strain()`.  Passing a boolean parameter in order to change the behavior of a function is generally to be avoided, as it makes the API harder to understand and the resulting code less readable.

Reply:
I also though about something like that. Since the stress is (in the case at hand) defined by 
```
stress = E * elast_strain = E * (total_strain - plast_strain)
```
a method `.elastic_stress()` is misleading in my opinion. I though about `.stress_elaststrain()`, `.stress_elastbehavior()`, `.stress_linear()` but since the result is just the stress I decided to go with one stress method. I agree, that the API is harder to understand with an optional parameter.

Reply:
Ok, so your point is, that the elastic stress for a given total strain is not the same thing as the stress for a given elastic strain, I see.  Is there a use case for this `.stress_for_elastic_strain()`?  If not, we should probably drop it as it really causes confusion, IMO.

@nae2rng, do you have an opinion on that?

Reply:
I agree with @johannes-mueller, the argument is confusing. Maybe a way to go is the following. We create an inherited class like so:
```python
class HookesLaw(RambergOsgood):
    def __init__(self, E):
        return super(RambergOsgood,self).__init__(E,K=0,n=0)
```
Then, the `stress` function could be overwritten to give `E * strain`. A function (or property) `hookes_law` could be added to `RambergOsgood`. The user would then do something like `rambgood.hookes_law.stress()`.

If that's too complicated, I'd recommend to drop the argument and maybe, if required, add a function with a clear naming.

Reply:
@johannes-mueller, I think it is useful to have the option to calculate the stress for a given elastic strain. Even if this calculation is simple in the case at hand, one would have to use the parameter `E` which was also used to instantiate to object in the first place. 

@nae2rng I like your idea of having a property `rambgood.hookes_law`, since it emphasises the fact that we are dealing with the linear elastic behaviour of the elastic-plastic material.

Reply:
Ok.  Even though I am not completely convinced about the use case, I think the solution with the derived class would be acceptable.  So to keep things simple and separated I would suggest to drop the parameter `eleasticstrain` in the `stress()` method.  The `HooksLaw` class can then be added in a second PR.